Looped/Authentication/SessionProvider.swift

```diff
@@ -18,23 +18,26 @@ final class SessionProvider: SessionProviding {
     
     private let sessionStore: SessionStoring
     private let disposeBag = DisposeBag()
+    private lazy var currentSessionSubject = BehaviorSubject<Session?>(value: sessionStore.getSession())
 
     // MARK: - SessionProvidingProtocol
     
+    var lastSession: Session?
+    
+    lazy var currentSession: Observable<Session?> = {
+        return currentSessionSubject
+    }()
+    
     func saveSession(session: Session) {
         lastSession = session
-        return sessionStore.save(session: session)
+        currentSessionSubject.onNext(session)
+        sessionStore.save(session: session)
```

Comment:
Tutaj powinieneś last session przypisać

Reply:
W sensie na końcu funkcji?
Btw, czemu w tych funkcjach miałem return, robiło to jakąś różnicę?

Reply:
Tak, na koncu funkcji. Zeby potem miec dosstep do tego latestSession gdybys potrzebowal

Reply:
Hmm może zły podgląd mi daje github, bo widzę funkcję saveSession, w której już robię lastSession = session?

A może chodzi Ci, żeby tutaj tak zrobić:
`lazy var currentSession: Observable<Session?> = {
        lastSession = try! currentSessionSubject.value()
        return currentSessionSubject
    }()`

Reply:
no nie wiem, to nie wyglada dobrze

Reply:
Mysle, ze lepiej byloby zrobic tak, ze session store ma `session` jako observabla i w inicie session store wywolac loadSession czy cos takiego, zeby ja zaladowac przy tworzeniu obiektu.

Reply:
`init(sessionStore: SessionStoring) {
        self.sessionStore = sessionStore
        lastSession = sessionStore.getSession()
    }`

A w ten sposób? I wtedy kasuję tę akcję z try! (...).value. Bo podoba mi się motyw
`private lazy var currentSessionSubject = BehaviorSubject<Session?>(value: sessionStore.getSession())`,
a jednocześnie nie wrzucałbym w ogóle Rxa do SessionStore - żeby miał 3 proste funkcję, get, save, delete